src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -142,6 +142,14 @@ protected int getNumOfElementOnPage(By cssSelectorBy) {
     return numElementOnPage;
   }
 
+  public void waitSafely(Runnable o) {
+    try {
+      o.run();
+    } catch (TimeoutException e) {

```

Comment:
actually it was me not understanding how [this rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166) works, fixed now